src/Cli/Program.cs

```diff
@@ -123,18 +122,43 @@ public static int Main(string[] args)
 
                     if (options.GraphQL)
                     {
-                        Task.Delay(1000).Wait();
-                        HttpClient client = new() { BaseAddress = new Uri("http://localhost:5000/graphql") };
-                        IntrospectionClient introspectionClient = new();
-                        Task<HotChocolate.Language.DocumentNode> response = introspectionClient.DownloadSchemaAsync(client);
-                        response.Wait();
-
-                        HotChocolate.Language.DocumentNode node = response.Result;
-                        File.WriteAllText(Path.Combine(options.OutputDirectory, options.GraphQLSchemaFile), node.ToString());
+                        int retryCount = 5;
+                        int tries = 0;
+
+                        while (tries < retryCount)
+                        {
+                            try
+                            {
+                                HttpClient client = new() { BaseAddress = new Uri("http://localhost:5000/graphql") };
+                                IntrospectionClient introspectionClient = new();
+                                Task<HotChocolate.Language.DocumentNode> response = introspectionClient.DownloadSchemaAsync(client);
+                                response.Wait();
+
+                                HotChocolate.Language.DocumentNode node = response.Result;
+
+                                if (!Directory.Exists(options.OutputDirectory))
+                                {
+                                    Directory.CreateDirectory(options.OutputDirectory);
+                                }
+
+                                string outputPath = Path.Combine(options.OutputDirectory, options.GraphQLSchemaFile);
+                                File.WriteAllText(outputPath, node.ToString());
```

Comment:
Does `WriteAllText` write Utf8 encoded text?

Reply:
Yes, it uses `UTF8NoBOM` (https://github.com/dotnet/dotnet/blob/main/src/runtime/src/libraries/System.Private.CoreLib/src/System/IO/StreamWriter.cs#L132) as the default encoding.